src/Pages/Admin.php

```diff
@@ -75,12 +75,21 @@ public function tlms_adminPanel(): void
     public function tlms_setupPage(): void
     {
         $action_status = $action_message = $api_validation = $domain_validation = '';
-        if (isset($_POST['action']) && $_POST['action'] == 'tlms-setup') {
-            if (isset($_POST['tlms-domain'], $_POST['tlms-apikey'], $_POST['tlms-enroll-user-to-courses'])
+
+        $action = isset($_POST['action']) ? sanitize_text_field(trim($_POST['action'])) : null;
+        $tlmsDomain = isset($_POST['tlms-domain']) ? sanitize_text_field(trim(strtolower($_POST['tlms-domain']))) : null;
+        $tlmsApiKey = isset($_POST['tlms-apikey']) ? sanitize_text_field(trim($_POST['tlms-apikey'])) : null;
+        $enrollUserToCourses = isset($_POST['tlms-enroll-user-to-courses']) ?
+            sanitize_text_field(trim($_POST['tlms-enroll-user-to-courses'])) : null;
+        $automaticallyCompleteOrders = isset($_POST['tlms-automtically-complete-orders']) ?
+            sanitize_text_field(trim($_POST['tlms-automtically-complete-orders'])) : null;
+
+        if (isset($action) && $action == 'tlms-setup') {
+            if (isset($tlmsDomain, $tlmsApiKey, $enrollUserToCourses)
             ) {
                 // we accept the domain only, without the protocol
-                if (stripos(strtolower($_POST['tlms-domain']), 'http') === 0
-                    || stripos(strtolower($_POST['tlms-domain']), 'https') === 0
+                if (stripos(strtolower($tlmsDomain), 'http') === 0
```

Comment:
You have already used `strtolower` for tlmsDomain and tlmsApiKey, I do not think we need it here.

Reply:
You are correct for the domain, not the tlmsApiKey, removed